### DISCUSSION TITLE
Enforce the maintenance on a given state

### DISCUSSION THREAD

**Discussion by NunoSH** [bmcclintock/momentuHMM#113] [Q&A]
Hi Brett,

quick question: is there any way to enforce the maintenance on the same state? 
For example, I have 5 states, and I want to enforce the maintenance on state 3 and 4 once each of these states start and have no say on the remaining transitions. So in essence, I think, what I want is to increase the probability on the non-transition in state 3 and 4 (or in the transition 3->3 and 4->4). 

As I have seen, we can influence the transitions from and to different states using "beta" on the Fixpar, or formula, but I am not sure if or how I could do the opposite.

I hope I made myself clear :)

Thank you very much!

Nuno

**Comment by bmcclintock** [comment]
I'm not sure I understand what you're wanting to do, but if you're trying to make both states 3 and 4 absorbing (i.e., once an individual enters state 3 or 4, then no more state switches can occur), then you can do this using the `fixPar` argument in `fitHMM`. For example,
```
fixPar <- list(beta=matrix(c(NA,NA,NA,NA,
                             NA,NA,NA,NA,
                             -1.e+100,-1.e+100,-1.e+100,-1.e+100,
                             -1.e+100,-1.e+100,-1.e+100,-1.e+100,
                             NA,NA,NA,NA),1,5*4))
```
would effectively prevent switches out of states 3 and 4.

**Reply by NunoSH** [reply]
Good morning Brett,

thanks a lot for your reply!

In a sense, it is more or less what you suggested, but with the possibility of switching out from these states. I would like to make it "harder" to switch out from these states but not impossible.

Because in order to an individual to get into state 3 or 4, there needs to be the presence (1-presence, 0-absence) of a feature (modeled with the Bernoulli distribution) and the probabilities of presence of these features (in state 3 and 4) have a really high values (>0.9). But what sometimes happens is that there is a punctual absence of these features, but I know the individuals are still in one of those states, but the HMM result gives that there is a switch out of one of these states, when actually there isn't. So I would like that these punctual absences of these features hadn't as much influence on switch off from state 3 or 4.

So to tackle this issue, is there a way of doing it in the MomentuHMM, or you thing it is better to address the data itself? (like doing a sort of smoothing or moving average for binary variables?

Thanks again!
I wish you a great day!

Nuno

**Comment by bmcclintock** [comment]
It's still hard for me to figure out what you're trying to do without much more detail. If you can provide a complete description of the data, model, and what you're trying to accomplish then I might be able to provide some suggestions. Code would also help. Based on your description, it sounds like you somehow know the individuals are still in state 3 or 4 even when there is a "punctual absence of these features".  I don't know what "punctual absence of these features" means, but, assuming this means they are either missing (i.e. `NA`) or `0`, what additional information do you have that indicates they must still be in state 3 or 4? This additional information would probably need to be incorporated into the model, either through additional (possibly constructed) covariates on the state transition probabilities or additional data streams.
